### PR TITLE
Add back --latest examples but only for local podman

### PR DESCRIPTION
**Pull request by rhatdan**
Also add --latest example to all commands that support it.

podman-remote does not support --latest option.

#### Does this PR introduce a user-facing change?

```release-note
--latest option now documented for all commands that support it in examples, not in podman-remote.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@baude PTAL

**Comment by rhatdan** [comment]
Follow on to https://github.com/containers/podman/pull/21177

File: cmd/podman/pods/stats.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ var (
 		ValidArgsFunction: common.AutocompletePodsRunning,
 		Example: `podman pod stats
   podman pod stats a69b23034235 named-pod
+  podman pod stats --latest
```
Comment (vrothberg) [inline cmd/podman/pods/stats.go:41]: I think this should be conditional.

File: cmd/podman/pods/inspect.go [hunk 2]
```diff
@@ -27,6 +27,9 @@ var (
 )
 
 func init() {
+	if !registry.IsRemote() {
+		inspectCmd.Example += "\n  podman pod inspect --latest"
+	}
```
Comment (TomSweeneyRedHat) [inline cmd/podman/pods/inspect.go:32]: Do you need a containerInspectCommand here too?  There are a few other commands that only have one line in this if, instead of 2 like the majority do.

Comment (rhatdan) [reply cmd/podman/pods/inspect.go:32]: These are pod commands not container commands.  Only containers commands can be called in two ways

podman COMMAND
podman container COMMAND

There is no equivalent for Pods, you always need to specify
podman pod COMMAND>

**Comment by Luap99** [comment]
IMO this is really not worth it, it adds additional complexity in our code for arguably no benefit.

We choose the way to pretend podman-remote (podman on win and macos) == podman on linux. If we continue to "advertise" local only options it means more and more users run into them on remote and end up being confused. It just needs someone to use it in a blog/tutorial and then remote users will run into `Error: unknown flag: --latest` which is really not helpful.
Because this is only a convenience option I do not see the point of showing this in the examples section.

**Comment by edsantiago** [comment]
I agree with Paul. What purpose does it serve to list `-l` in examples? Document what works everywhere. If some user knows about `-l`, they can play with it and see if it works. We don't need to encourage it.

**Comment by rhatdan** [comment]
We documented it before, it is a useful feature which all of us enjoy, I believe.  Now we want to hide it from the users who can take advantage.

@baude WDYT?

**Comment by vrothberg** [comment]
> Now we want to hide it from the users who can take advantage.

I don't think the desire is to hide it since `--latest` will continue show up in the `--help` message by being listed as a flag; at least for local clients.

**Comment by Luap99** [comment]
I don't want to hide the flag from the local podman --help and I have no issues with the current docs as they say it is local only.

The issue is that a random example does not say it is local only (and many macos/windows users would not know what it means). Basically every time someone (including us) uses local only options/commands in examples/tutorials/blogs,etc... we will cause issues for remote client users (mostly on macos and windows) as they will run into a non working options/command and end up being confused and then either report bugs like https://github.com/containers/podman/issues/21174 or potentially just give up on podman as tutorials or docs are in their opinion incorrect/not working.

**Comment by rhatdan** [comment]
If this PR is closed, then --latest will not be documented in podman COMMAND --help EXAMPLES commands, so it is becoming more hidden.

Whether we use --latest or other non --remote commands in Blogs needs to be guarded, but I don't see how this PR effects whether or not someone uses the option in a public site.  

As far as Examples in Man pages, I took care of that in a separate PR, but we have on ability yet to guard against new examples being added with --latest.

**Comment by edsantiago** [comment]
I'm having a lot of trouble understanding this thread. I've slept on it, and am fresh-brained (well, for me), and I just don't see how "not including -l in help-message examples" translates to "more hidden". And I don't understand why we even care? To my mind the examples in `--help` should give one or two representative illustrations of the ways one would use a command. It can't possibly be every permutation.

Full disclosure though: I loathe `-l` and have never used it. It is not reliable in my multi-terminal workflow with periodic cron jobs that run podman. I consider `-l` a recipe for chaos. And with that said, why exactly does `-l` not work with remote? My understanding is "because connection could be from different user", but in reality what I suspect is happening with Mac/Windows is single-seat single-workflow single-thread command typing. 

TL;DR can the "-l doesn't work remote" decision be reexamined for podman 5?

**Comment by Luap99** [comment]
All API endpoints work with names or ids `/containers/{id}/json`,etc... So there is no way to express the latest option via the API, so we either have to support that in each API endpoint which seems broken as you said this will not work reliably and should really never be used in an API. The other option would be to get the lastest container in podman-remote via container list endpoint and then feed the ID to all the API calls but this also has big TOCTOU race conditions (the local one does too of course but it is much smaller as things do not have to be send over the wire).

**Comment by rhatdan** [comment]
Since @baude did not come to argue for this, closing.
